docs/hub/how-to-downstream.md

```diff
@@ -0,0 +1,55 @@
+# How to integrate downstream methods in your library
+
+In the tutorial, you learned some basic methods for integrating the Hub into your library. Methods that allow your library to download files from the Hub are referred to as *downstream* methods. This guide introduces additional downstream methods you can integrate in your library. You will learn how to:
+
+* Retrieve a URL to download.
+* Download a file and cache it on your disk.
+* Download all the files in a repository.
+
+## `hf_hub_url`
+
+Use `hf_hub_url` to retrieve the URL of a specific file to download by providing a `filename`.
+
+![/docs/assets/hub/repo.png](/docs/assets/hub/repo.png)
+
+```python
+>>> from huggingface_hub import hf_hub_url
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json")
+'https://huggingface.co/lysandre/arxiv-nlp/resolve/main/config.json'
+```
+
+Specify a particular file version by providing the file revision. The file revision can be a branch, a tag, or a commit hash. If you choose to use the commit hash, it must be the full-length hash instead of a 7-character commit hash.
+
+```python
+>>> hf_hub_url(repo_id="lysandre/arxiv-nlp", filename="config.json", revision="877b84a8f93f2d619faa2a6e514a32beef88ab0a")
```

Comment:
Would it be helpful to show an example involving a branch and tag? (e.g. on the same `lysandre/arxiv-nlp` repo so the reader can verify for themselves what that looks like in practice)